src/common/utils/crypto.utils.ts

```diff
@@ -0,0 +1,29 @@
+import { randomBytes, scrypt, timingSafeEqual } from 'crypto'
+
+const keyLength = 64
+
+export const hash = async (password: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
```

Comment:
No need for custom Promise construction. Please use `promisify` instead:
```TS
import { promisify } from 'node:util';
```

Reply:
Fixed